explainability-core/src/test/java/org/kie/trustyai/metrics/drift/fouriermmd/FourierMMDTest.java

```diff
@@ -41,6 +43,44 @@ protected void setup() throws Exception {
         testDF = readCSV(testDataSetFileName);
     }
 
+    protected static Dataframe generateRandomDataframe(int observations, int featureDiversity) {
+        final List<Prediction> predictions = new ArrayList<>();
+        final Random random = new Random(0);
+        for (int i = 0; i < observations; i++) {
+            final List<Feature> featureList = List.of(
+                    // guarantee feature diversity for age is min(observations, featureDiversity)
+                    FeatureFactory.newNumericalFeature("age", i % featureDiversity),
+                    FeatureFactory.newNumericalFeature("gender", random.nextBoolean() ? 1 : 0),
+                    FeatureFactory.newNumericalFeature("race", random.nextBoolean() ? 1 : 0));
+            final PredictionInput predictionInput = new PredictionInput(featureList);
+
+            final List<Output> outputList = List.of(
+                    new Output("income", Type.NUMBER, new Value(random.nextBoolean() ? 1 : 0), 1.0));
+            final PredictionOutput predictionOutput = new PredictionOutput(outputList);
+            predictions.add(new SimplePrediction(predictionInput, predictionOutput));
+        }
+        return Dataframe.createFrom(predictions);
+    }
+
+    protected static Dataframe generateRandomDataframeDrifted(int observations, int featureDiversity) {
+        final List<Prediction> predictions = new ArrayList<>();
+        final Random random = new Random(0);
+        for (int i = 0; i < observations; i++) {
+            final List<Feature> featureList = List.of(
+                    // guarantee feature diversity for age is min(observations, featureDiversity)
+                    FeatureFactory.newNumericalFeature("age", (i % featureDiversity) + featureDiversity),
+                    FeatureFactory.newNumericalFeature("gender", 0),
+                    FeatureFactory.newNumericalFeature("race", random.nextBoolean() ? 1 : 0));
+            final PredictionInput predictionInput = new PredictionInput(featureList);
+
+            final List<Output> outputList = List.of(
+                    new Output("income", Type.NUMBER, new Value(random.nextBoolean() ? 1 : 0), 1.0));
+            final PredictionOutput predictionOutput = new PredictionOutput(outputList);
+            predictions.add(new SimplePrediction(predictionInput, predictionOutput));
+        }
+        return Dataframe.createFrom(predictions);
+    }
+
```

Comment:
import this from `MockDatasource` rather than copy it

Reply:
I tried to do this initially:

<img width="622" alt="image" src="https://github.com/trustyai-explainability/trustyai-explainability/assets/13667247/127dc791-8dc1-4b3d-976c-4305232852ea">

Seems like "service" is not included in the test compile?

Reply:
Yeah, you're right, can't import anything from service to core, okay, leave it as is.